src/thor/route_matcher.cc

```diff
@@ -54,6 +54,30 @@ float length_comparison(const float length, const bool exact_match) {
   return length + tolerance;
 }
 
+// check if the intermediate shape points are also on the edge
+bool MatchShape(const graph_tile_ptr& tile,
```

Comment:
nit, casing should be the same as the other methods, also the term `match` is used so much perhaps `check` is better for something that is a helper method:

```suggestion
bool check_shape(const graph_tile_ptr& tile,
```